src/twisted/trial/_dist/worker.py

```diff
@@ -68,20 +79,44 @@ class WorkerProtocol(AMP):
     The worker-side trial distributed protocol.
     """
 
+    logger = Logger()
+
     def __init__(self, forceGarbageCollection=False):
         self._loader = TestLoader()
         self._result = WorkerReporter(self)
         self._forceGarbageCollection = forceGarbageCollection
 
     @workercommands.Run.responder
-    def run(self, testCase):
+    async def run(self, testCase: str) -> RunResult:
         """
         Run a test case by name.
         """
-        case = self._loader.loadByName(testCase)
-        suite = TrialSuite([case], self._forceGarbageCollection)
-        suite.run(self._result)
-        return {"success": True}
+        with self._result.gatherReportingResults() as results:
+            case = self._loader.loadByName(testCase)
+            suite = TrialSuite([case], self._forceGarbageCollection)
+            suite.run(self._result)
+
+            allSucceeded = True
+            for (success, result) in await DeferredList(results, consumeErrors=True):
+                if not success:
+                    allSucceeded = False
```

Comment:
Maybe reduce the indentation and add explicit handling of the success case.

```suggestion
            for (success, result) in await DeferredList(results, consumeErrors=True):
                if success:
                    # Test execution was successful and there is no need for extra
                    # reporting.
                    continue

                # Report the failed test.
                allSucceeded = False
```